conan/tools/cmake/toolchain.py

```diff
@@ -535,14 +535,12 @@ def _get_toolset(self, generator):
             return IntelCC(self._conanfile).ms_toolset
         elif compiler == "msvc":
             compiler_version = str(settings.compiler.version)
-            version_components = compiler_version.split(".")
-            if len(version_components) >= 2:  # there is a 19.XX
-                minor = version_components[1]
-                if len(minor) >= 2 and "X" not in minor:  # It is full one(19.28), not generic 19.2X
-                    # The equivalent of compiler 19.26 is toolset 14.26
-                    return "version=14.{}".format(minor)
-                else:
-                    return "v14{}".format(minor[0])
+            compiler_update = str(settings.compiler.update)
+            if compiler_update != "None":  # It is full one(19.28), not generic 19.2X
+                # The equivalent of compiler 19.26 is toolset 14.26
+                return "version=14.{}{}".format(compiler_version[-1], compiler_update)
```

Comment:
This does not correctly handle special cases for VS 16.9 and VS 16.11, which need 3-part version number to be selected correctly. See [this thread](https://developercommunity.visualstudio.com/t/How-to-select-VS-168-compiler-with-vcva/1359197) for more information.

Reply:
This piece of code will handle whatever you have defined in your ``msvc`` ``compiler.update``. If you define your updates as ``9.XX`` and ``9.YY`` (which you can easily add to the default ``settings.yml``) to differentiate those different cases, this would work wouldn't it?

Reply:
🤔  - it should work. Excellent idea! So, for VS 16.8 I would need to add "8.29333" and for VS 16.9 "9.30037". 

Seems elegant enough to even be part of the default `settings.yml` 😊